ci/setup_cmake.sh

```diff
@@ -5,9 +5,17 @@ set -e
 export DEBIAN_FRONTEND=noninteractive
 apt-get update
 
+export CMAKE_VERSION=3.15.2
+
+pushd /tmp
+wget https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.sh
```

Comment:
I have a very similar script here. Main benefit it also validates what version you get, so it doesn't download if you already have a good version: https://github.com/open-telemetry/opentelemetry-cpp/blob/master/tools/setup-cmake.sh

It is cross-platform, but it builds from source. Also it does build it without ninja because on some older platforms that I tested, e.g. Ubuntu 14.x - ninja build of CMake fails. My main experience with building new CMake from source is that this entire process is rather slow. If we download / build it for some platform, we should cache it in github action cache.

Reply:
Basically this upgrade would add 9 minutes to `CI / CMake test (with otlp-exporter)` run.

Reply:
The current script is mostly for CI and may also help repro CI related issue locally. I think it is unnecessary to rebuild CMake from source in this restricted environment, and do it for every CI build.

Adding a version check sounds like a good improvement we could do in later PR.